lib/nuget-parser/dotnet-core-v2-parser.ts

```diff
@@ -0,0 +1,168 @@
+import * as debugModule from 'debug';
+import * as depGraphLib from '@snyk/dep-graph';
+import { DepGraphBuilder } from '@snyk/dep-graph';
+
+const debug = debugModule('snyk');
+
+interface ProjectAssets {
+  project: any;
+  targets: any;
+}
+
+interface DotnetPackage {
+  type: string;
+  dependencies?: Record<string, string>;
+  runtime?: object;
+  compile: object;
+  // Not from assets.json, but injected during building of graph for easy access.
+  name: string;
+  version: string;
+}
+
+export interface Assembly {
+  name: string;
+  version: string;
+}
+
+export type AssemblyVersions = Record<string, string>;
+
+// Dependencies that starts with these are discarded
+const FILTERED_DEPENDENCY_PREFIX = ['runtime'];
```

Comment:
These pollute the generated dependency graph, and are (to the best of my understanding) not relevant until we start allowing users to specify `TargetRuntimeIdentifier` in their projects.